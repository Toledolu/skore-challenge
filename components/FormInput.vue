<template>
  <input class="" v-on="inputListeners"
    :class="classes"
    :type="type"
    :placeholder="placeholder"
    :value="value"
  >
</template>

<script>
export default {
  name: 'FormInput',
  props: {
    classes: {
      type: String,
      default: ''
    },
    value: {
      type: [String, Number],
      default: undefined,
    },
    placeholder: {
      type: String,
      default: undefined,
    },
    type: {
      type: String,
      default: 'text',
    },
  },
  data() {
    return {
      localValue: this.value,
    };
  },
  watch: {
    value(value) {
      this.localValue = value;
    },
  },
  computed: {
    inputListeners() {
      const vm = this;
      return {
        ...this.$listeners,
        input(event) {
          vm.$emit('input', event.target.value);
        },
      };
    },
  },
};
</script>