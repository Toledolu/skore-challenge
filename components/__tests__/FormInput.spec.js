import { shallowMount } from "@vue/test-utils";
import FormInput from '../FormInput.vue';


const factory = () => {
  return shallowMount(FormInput, {
  });
};

describe("FormInput", () => {
  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});