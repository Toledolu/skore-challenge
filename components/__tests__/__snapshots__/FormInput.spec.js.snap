// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput renders properly 1`] = `"<input type="text" class="">"`;
